enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,243 @@
+
+For example: Given the below configuration, the system will keep at most 1 rotated log file. If that log file reaches >7 days old, it will be deleted, regardless of whether the live log has reached the max file size of 200Mb.  Otherwise, if the live log reaches the 200Mb limit, it will be rotated, causing the existing log backup (if any) to be deleted. 
+```yaml
+apiServer:
+  auditLog:
+    maxFileSize: 200
+    maxFiles: 1
+    maxFileAge: 7
+  ```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+- N/A
+
+#### Standalone Clusters
+
+- N/A
+
+#### Single-node Deployments or MicroShift
+
+- This EP is specific to MicroShift only.
+
+### Implementation Details/Notes/Constraints
+
+**Passing User Provided Options to the API Server**
+
+MicroShift embeds a default API server config, written to disk during boot at `/var/lib/microshift/resources/kube-apiserver-audit-policies/default.yaml`. The file is structured as a `KubeAPIServerConfig` custom resource, provided by the `kubecontrolplane.config.openshift.io/v1` API group. During boot, the kube-apiserver service-manager collects kube-apiserver options from hardcoded defaults and merges these into the kube-apiserver config data structure.  The override values are iteratively patch onto the embedded default config.  The order of precedence is from first to last, such that field paths appearing later in the queue override data fields from previous iterations.  The logic which handles the overriding is [here](https://github.com/openshift/microshift/blob/ba79f993070111ae3b6e35c3abb3141e95901874/pkg/controllers/kube-apiserver.go#L226-L250). Thus, to pass user-defined audit configuration options to the API server, MicroShift will append the user-specified options to `overrides` array. This minimizes the complexity of the option-passing and allows for MicroShift defaults to continue to be specified as they are now.  This also provides an elegant fallback behavior for user-facing fields. If a user does not specify a value for a field, the default value will be used, just as it is now.  Should a user remove a field they previously set, then at next restart, the field will no longer be overridden, thus passing the default. All audit logging configuration fields must be optional. If not specified, the existing default audit logging policy will be used.
+
+**Default Values**
+
+MicroShift currently sets default values for `audit-log-maxbackup` to "10" and `audit-log-maxsize` to "200". Thus default log storage is 2,000 Mb. These values are defined as part of the embedded [kube-apiserver config file](https://github.com/openshift/microshift/blob/66394512e6adac09f26e4cf049951d77a83a05a4/assets/controllers/kube-apiserver/defaultconfig.yaml#L44-L48). This design will not add a default value for `audit-log-maxage`.  It is sufficient to specify a storage limit on logs for default deployments. 
```

Comment:
This might be overly pedantic but I think we need to pay attention to Mb (megabits) vs MB (megabytes). I guess whenever document uses Mb it means megabytes but I can at least imagine someone scratching their head and doing multiply/divide by 8 math.